solr/server/contexts/solr-jetty-context.xml

```diff
@@ -5,4 +5,19 @@
   <Set name="war"><Property name="jetty.base"/>/solr-webapp/webapp</Set>
   <Set name="defaultsDescriptor"><Property name="jetty.base"/>/etc/webdefault.xml</Set>
   <Set name="extractWAR">false</Set>
+  <Set name="parentLoaderPriority">true</Set>
+  <Call name="addServerClassMatcher">
+    <Arg>
+      <New id="removeServerClasses" class="org.eclipse.jetty.webapp.ClassMatcher">
```

Comment:
What's this about?

Reply:
Yea agree - would be good to add a comment to why this is needed?

Reply:
Yeah will definitely add a comment. But here's the gist:

https://www.eclipse.org/jetty/documentation/jetty-9/index.html#jetty-classloading

Jetty protects access to certain "server" classes. You can see the defaults if you scroll down to the "Setting Server Classes" section in the documentation link above. (Annoyingly the same documentation doesn't exist for Jetty 10 or 11, although the functionality is the same, other than a method signature) This means that the webapp cannot use these classes unless they are also provided in the WEB-INF/lib directory as well, meaning that we have to duplicate these jars using the default settings.

However when we add this set of `serverClassMatchers`, which all start with `-`, it removes these classes from the protected "server" class definition. This allows us to use the same jars between the jetty server and our webapp.

The documentation doesn't actually give any good reasons why these classes should be protected, or maybe I just missed it. So we can always go back to duplicating the jars, but I think this is probably a better way forward.

Obviously we need some pretty extensive testing to make sure that the `setParentLoaderPriority​(true)` option doesn't mess with any of our class loader stuff. So far the integration tests, docker tests and unit tests all work so far. But we will definitely need to do some more.

Reply:
Thanks for that explanation; please add a summary as a comment in the XML.
I definitely prefer a solution (like this) that avoids duplicate JARs; I didn't know we were doing that (yuck!).

RE classpath separation (Jetty vs web app), note there is a nasty line in bin/solr when SOLR_SSL_ENABLED:
```
SOLR_JETTY_CONFIG+=("--module=https" "--lib=$DEFAULT_SERVER_DIR/solr-webapp/webapp/WEB-INF/lib/*")
```
I recall the original introduction of this was by @markrmiller when I did git history analysis some time ago.  I feel we should either not do that, or just give up on the separation -- this is a single-web app server so why separate?

Reply:
> RE classpath separation (Jetty vs web app), note there is a nasty line in bin/solr when SOLR_SSL_ENABLED:

I have a sneaking suspicion that isn't necessary, at least anymore. We should try to tackle that in a separate PR though. Would like to not have to test SSL at the same time 😅